src/spent_proof.rs

```diff
@@ -1,56 +1,57 @@
+// Copyright 2022 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
 use crate::{
-    Dbc, Error, Hash, KeyManager, NodeSignature, PublicKey, PublicKeySet, Result, Signature,
+    Commitment, Error, Hash, KeyImage, KeyManager, NodeSignature, PublicKey, PublicKeySet, Result,
+    Signature,
 };
 
-use serde::{Deserialize, Serialize};
-
-#[derive(Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize)]
-pub struct SpendKey(pub PublicKey);
-
-// Display Hash value as hex in Debug output.  consolidates 36 lines to 3 for pretty output
-// and the hex value is the same as sn_dbc_mint display of DBC IDs.
-impl std::fmt::Debug for SpendKey {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.debug_tuple("SpendKey")
-            .field(&hex::encode(self.0.to_bytes()))
-            .finish()
-    }
-}
-
-impl SpendKey {
-    pub fn to_bytes(self) -> [u8; 48] {
-        self.0.to_bytes()
-    }
-}
+use std::cmp::Ordering;
+use std::hash;
 
-#[cfg(test)]
-use rand::distributions::{Distribution, Standard};
-#[cfg(test)]
-use rand::Rng;
-
-/// used when fuzzing DBC's in testing.
-#[cfg(test)]
-impl Distribution<SpendKey> for Standard {
-    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> SpendKey {
-        SpendKey(
-            crate::genesis_dbc_input()
-                .0
-                .derive_child(&rng.gen::<[u8; 32]>()),
-        )
-    }
-}
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
 
 /// A share of a SpentProof, combine enough of these to form a
 /// SpentProof.
-#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Debug, Clone)]
 pub struct SpentProofShare {
-    /// Signature from dbc.spend_key() over the transaction
-    pub spent_sig: Signature,
+    pub key_image: KeyImage,
 
     /// The Spentbook who notarized that this DBC was spent.
     pub spentbook_pks: PublicKeySet,
 
     pub spentbook_sig_share: NodeSignature,
+
+    pub public_commitments: Vec<Commitment>,
```

Comment:
We need to be including the public_commitments in the spent proof signature. Otherwise mint's can't trust that the public commitments here were checked for validity by the spent book nodes

Reply:
ok, this is a bit more involved and I think requires mods in spentbook.rs and also validation.rs.    leaving for tmw.

Reply:
ok, I made a SpentProofContent to hold all the data that SpentBook needs to sign, and it now holds the transaction hash.
```
pub struct SpentProofContent {
    pub key_image: KeyImage,
    pub transaction_hash: Hash,
    pub public_commitments: Vec<Commitment>,
}
```
So `SpentProofShare` and `SpentProof` both hold a `SpentProofContent` plus signing info.

confirm_valid() also verifies that the incoming tx hash matches this one.  And did some cleanup/refactoring in spent_proof.rs.  So I think it's good now.

commit 59d7f137bb5275b50dd7c6c4aedd06ee5cc85702